shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp

```diff
@@ -9,81 +9,51 @@
 #include <limits>
 
 #include "backend/protobuf/transaction_responses/proto_concrete_tx_response.hpp"
-#include "cryptography/hash.hpp"
 
 namespace shared_model {
   namespace proto {
     /**
      * TransactionResponse is a status of transaction in system
      */
-    class TransactionResponse final
-        : public CopyableProto<interface::TransactionResponse,
-                               iroha::protocol::ToriiResponse,
-                               TransactionResponse> {
+    class TransactionResponse final : public interface::TransactionResponse {
      public:
-      /// Type of variant, that handle all concrete tx responses in the system
-      using ProtoResponseVariantType =
-          boost::variant<StatelessFailedTxResponse,
-                         StatelessValidTxResponse,
-                         StatefulFailedTxResponse,
-                         StatefulValidTxResponse,
-                         RejectedTxResponse,
-                         CommittedTxResponse,
-                         MstExpiredResponse,
-                         NotReceivedTxResponse,
-                         MstPendingResponse,
-                         EnoughSignaturesCollectedResponse>;
-
-      /// Type with list of types in ResponseVariantType
-      using ProtoResponseListType = ProtoResponseVariantType::types;
-
-      template <typename TxResponse>
-      explicit TransactionResponse(TxResponse &&ref);
+      using TransportType = iroha::protocol::ToriiResponse;
 
       TransactionResponse(const TransactionResponse &r);
-
       TransactionResponse(TransactionResponse &&r) noexcept;
 
+      explicit TransactionResponse(const TransportType &ref);
+      explicit TransactionResponse(TransportType &&ref);
```

Comment:
Why already existing constructors (above) are not marked with explicit?

Reply:
https://stackoverflow.com/a/6759059

TL;DR - copy and move constructors are expected to be nonexplicit